src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -36,192 +412,4080 @@ Position const SylvanasIntroPos[4] =
+
+                        if (IsArrowPositionValid(middleLineFront, 60, additionalMilliseconds))
+                            arrowPositions.push_back(middleLineFront);
+
+                        Position middleLineBack(pos.GetPositionX() + (std::cos(orientation + -90.0f * float(M_PI) / 180) * pointDistance), pos.GetPositionY() + (std::sin(orientation + -90.0f * float(M_PI) / 180) * pointDistance), pos.GetPositionZ());
+
+                        if (IsArrowPositionValid(middleLineBack, 60, additionalMilliseconds))
+                            arrowPositions.push_back(middleLineBack);
+                    }
+                    else
+                    {
+                        Position leftLineFront(pos.GetPositionX() + (std::cos(orientation + 90.0f * float(M_PI) / 180) * pointDistance), pos.GetPositionY() + (std::sin(orientation + 90.0f * float(M_PI) / 180) * pointDistance), pos.GetPositionZ());
+
+                        if (IsArrowPositionValid(leftLineFront, 70, additionalMilliseconds))
+                            arrowPositions.push_back(leftLineFront);
+
+                        Position leftLineBack(pos.GetPositionX() + (std::cos(orientation + -90.0f * float(M_PI) / 180) * pointDistance), pos.GetPositionY() + (std::sin(orientation + -90.0f * float(M_PI) / 180) * pointDistance), pos.GetPositionZ());
+
+                        if (IsArrowPositionValid(leftLineBack, 70, additionalMilliseconds))
+                            arrowPositions.push_back(leftLineBack);
+                    }
+                }
+
+                // Note: after hours of research and Pitagoras, each part of the arrow has a different travelSpeed.
+                float travelSpeed = 0.245000004768371f;
+
+                for (uint8 i = 0; i < arrowPositions.size(); ++i)
+                {
+                    if (npc_sylvanas_windrunner_shadowcopy* shadowCopyAI = GetShadowcopyCastAI(instance, amount))
+                        shadowCopyAI->StoreDesecratingShots(arrowPositions[i], step, travelSpeed + float(travelSpeed * (step * 0.025)));
+                }
+
+                return true;
+            }
+
+            case DATA_DESECRATING_SHOT_PATTERN_SPIRAL:
+            case DATA_DESECRATING_SHOT_PATTERN_SPIRAL_ENCLOSED:
+            {
+                float distance = pattern == DATA_DESECRATING_SHOT_PATTERN_SPIRAL ? DesecratingShotNormalSpiralDistance[step - 1] : DesecratingShotSpiralEnclosedDistance[step - 1];
+
+                if (step != 1)
+                {
+                    Position spiralLeft(pos.GetPositionX() + (std::cos((orientation + 120.0f + float(10 * step)) * float(M_PI) / 180) * distance), pos.GetPositionY() + (std::sin((orientation + 120.0f + float(10 * step)) * float(M_PI) / 180) * distance), pos.GetPositionZ());
```

Comment:
move the orientation calculation to separate variable except the changing value (120, 240, 360)